includes/connect/class-wc-stripe-connect.php

```diff
@@ -131,7 +131,7 @@ public function connect_oauth( $state, $code, $type = 'connect', $mode = 'live'
 				return $response;
 			}
 
-			delete_transient( 'wcs_stripe_connect_state_' . $mode );
+			WC_Stripe_Database_Cache::delete( 'wcs_stripe_connect_state_' . $mode );
```

Comment:
As above, I am concerned that the current mode (in the `testmode` setting) may not match the mode at the time the write occurred.